app/models/employee.rb

```diff
@@ -412,21 +406,19 @@ def create_payslip(employee, salary_date)
     total_salary -= tot_deduction.to_f
     b = MonthlyPayslip.where(employee_id: employee.id, salary_date: salary_date).pluck(:salary_date)
     if b[0].present?
-      if b[0] == salary_date.strftime('%b')
-        flag = 1
-    end
+      flag = 1 if b[0] == salary_date.strftime('%b')
     else
       MonthlyPayslip.create(salary_date: salary_date, employee_id: employee.id, amount: total_salary)
       flag = 0
-       end
+    end
     flag
   end
 
   # This method is used for create or update employee salary strcture
   # if salary structure already created then update it otherwise update
   def self.emp(emp, payroll, amount)
-    salary = EmployeeSaleryStructure.where(employee_id: emp.id,
-                                           payroll_category_id: payroll).take
+    salary = EmployeeSaleryStructure.find_by(employee_id: emp.id,
```

Comment:
you might want to test the impact of this line properly as the returned outputs of `where` & `find_by` method are different.